src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -605,7 +632,7 @@ internal void Configure()
             PropertyInfoForTypeInfo.EnsureChildOf(this);
             PropertyInfoForTypeInfo.EnsureConfigured();
 
-            CanUseSerializeHandler &= Options.CanUseFastPathSerializationLogic;
+            CanUseSerializeHandler &= OriginatingContext?.CanUseFastPathSerializationLogic(Options) == true;
```

Comment:
Will this work correctly for all cases? See scenarios below. With this check seems this will return true. Shouldn't we first check if all combined resolvers are contexts with compatible options first? And in such case we shouldn't need new API because we could put that internal property on the combined context directly

Scenarios:
- Foo contract is source-gened, Foo contains property Bar, Bar is customized by different resolver
- source gen creates JsonTypeInfo but we have a wrapping resolver which customizes it - the originating resolver will still be JsonSerializerContext even though it's the wrapping one which made last modification